index.js

```diff
@@ -3,105 +3,175 @@ import got from 'got';
+                        error: e.message || JSON.stringify(e)
+                    };
                 }
-              }
-            );
-            return [];
-          } catch (e) {
-            return [{
-              id,
-              tier,
-              file: FileName,
-              error: JSON.stringify(e)
-            }];
-          }
-        })
-      )
-    ));
-  return errors.flatMap((a) => a);
+            })
+        )
+    );
+
+    // Filter out nulls to return only the errors
+    return errors.filter(e => e !== null);
 }
 
+
+/**
+ * Main execution function.
+ */
 async function main() {
-  const {
-    tier,
-    folder,
-    secret,
-    from_id,
-    parallel,
-    progress // This will be true if '--progress' is passed, otherwise undefined
-  } = minimist(process.argv.slice(2));
-
-  assert(folder, `Required arg: folder`);
-  assert(secret, `Required arg: secret. Look for the Authorization header in a download request`);
-  const maxTier = tier ?? 10;
-  const minId = from_id ?? 0;
-
-  const result = await got.get(`https://api.printableheroes.com/api/minis/getAll`, {
-    headers: {
-      'Content-Type': 'application/json; charset=utf-8',
-    },
-  });
-
-  const limit = pLimit(parallel ?? 5);
-  const ids = JSON.parse(result.body).filter(({
-    Id
-  }) => Id >= minId).map(({
-    Name,
-    Id
-  }) => [Name.trim(), Id.toString().trim()]).sort((a, b) => a[1] - b[1]);
-
-  console.log(ids.length === 0 ? `No new heroes since {${minId}}` : `Downloading {${ids.length}} heroes...`);
-
-  if (ids.length > 0) {
-    let progressBar;
-    if (progress) {
-      // If --progress is used, create a real progress bar
-      progressBar = new cliProgress.SingleBar({}, cliProgress.Presets.shades_classic);
-    } else {
-      // Otherwise, create a dummy object with empty functions so the rest of the code works without changes.
-      progressBar = {
-        start: () => {},
-        increment: () => {},
-        stop: () => {},
-      };
+    const {
+        tier,
+        folder,
+        secret,
+        from_id,
+        parallel
+    } = minimist(process.argv.slice(2));
+
+    // Assert that required arguments are provided
+    assert(folder, `Required arg: --folder <path>`);
+    assert(secret, `Required arg: --secret <token>. Look for the Authorization header in a download request.`);
+
+    // Set default values for optional arguments
+    const maxTier = tier ?? 10;
+    const minId = from_id ?? 0;
+    const concurrency = parallel ?? 5;
+
+    // Fetch the list of all miniatures
+    const result = await got.get(`https://api.printableheroes.com/api/minis/getAll`, {
+        headers: {
+            'Content-Type': 'application/json; charset=utf-8',
+        },
+        responseType: 'json'
+    });
+
+    const limit = pLimit(concurrency);
+    const ids = result.body
+        .filter(({
+            Id
+        }) => Id >= minId)
+        .map(({
+            Name,
+            Id
+        }) => [
```

Comment:
Not a blocker, just generic perf advice: filter followed by map map can be done in a single pass using flatMap.

```typescript
if (/*clause*/) { [/ *map */ ] } else { [] }
```